src/app/budget/[month]/components/TransactionTable.tsx

```diff
@@ -0,0 +1,283 @@
+"use client";
+import { useEffect, useState } from "react";
+import { Transaction } from "@/types/budget";
+import { transactionService } from "@/lib/transactionService";
+import { BUDGET_CATEGORIES } from "@/constants/budget";
+import { eventBus } from "@/utils/eventBus";
+import { BUDGET_EVENTS } from "@/utils/eventTypes";
+
+export function TransactionTable() {
+  const [transactions, setTransactions] = useState<Transaction[]>([]);
+  const [editingId, setEditingId] = useState<string | null>(null);
+  const [editingTransaction, setEditingTransaction] =
+    useState<Transaction | null>(null);
+
+  useEffect(() => {
+    const deleteSubscription = eventBus.on(
+      BUDGET_EVENTS.TRANSACTION_DELETED,
+      ({ transactionId }) => {
+        setTransactions((current) =>
+          current.filter((t) => t.id !== transactionId)
+        );
+      }
+    );
+
+    const updateSubscription = eventBus.on(
+      BUDGET_EVENTS.TRANSACTION_UPDATED,
+      ({ transactionId, updates }) => {
+        setTransactions((current) =>
+          current.map((t) =>
+            t.id === transactionId ? { ...t, ...updates } : t
+          )
+        );
+      }
+    );
+
+    const addSubscription = eventBus.on(
+      BUDGET_EVENTS.TRANSACTION_ADDED,
+      ({ transaction }) => {
+        setTransactions((current) => [...current, transaction]);
+      }
+    );
+
+    return () => {
+      deleteSubscription();
+      updateSubscription();
+      addSubscription();
+    };
+  }, []);
+
+  useEffect(() => {
+    const params = new URLSearchParams(window.location.search);
+    const month = window.location.pathname.split("/budget/")[1];
+
+    async function loadTransactions() {
+      const transactions = await transactionService.getTransactions(month);
+      setTransactions(transactions);
+    }
+
+    loadTransactions();
+  }, []);
+
+  const handleEdit = (transaction: Transaction) => {
+    setEditingId(transaction.id);
+    setEditingTransaction({ ...transaction });
+  };
+
+  const handleUpdate = async () => {
+    if (!editingTransaction || !editingId) return;
+
+    try {
+      await transactionService.updateTransaction(editingId, editingTransaction);
+
+      // Emit event to update UI
+      eventBus.emit(BUDGET_EVENTS.TRANSACTION_UPDATED, {
+        transactionId: editingId,
+        updates: editingTransaction,
+      });
+
+      // Reset editing state
+      setEditingId(null);
+      setEditingTransaction(null);
+
+      // Update transactions list
+      const month = window.location.pathname.split("/budget/")[1];
```

Comment:
oh, it scared me! :)  how will you maintain such code in future?)  window.location.pathname.split("/budget/")[1]; - not the best solution - use specific fabrics or add routing service to define url params for each page. It will make your code 100x times robust!